src/components/measure/MeasureService.js

```diff
@@ -0,0 +1,264 @@
+      if (km <= 0) {
+        return cleanAfterComma(measure) + units[1];
+      }
+
+      var str = '' + km;
+      var m = Math.floor(Math.floor(measure) % factor * 100 / factor);
+
+      if (m > 0) {
+        str += '.';
+        if (m < 10) {
+          str += '0';
+        }
+        str += m;
+      }
+      str += units[0];
+      return str;
+    };
+  });
+
+  module.provider('gaMeasure', function() {
+    this.$get = function($document, measureFilter) {
+      var Measure = function() {
+
+        this.getLength = function(geom) {
+          var lineString;
+          if (geom instanceof ol.geom.LineString) {
+            lineString = geom;
+          } else if (geom instanceof ol.geom.LinearRing) {
+            lineString = new ol.geom.LineString(geom.getCoordinates());
+          } else if (geom instanceof ol.geom.Polygon) {
+            lineString = new ol.geom.LineString(
+                geom.getLinearRing(0).getCoordinates());
+          } else if (geom instanceof ol.geom.Circle) {
+            return 2 * Math.PI * geom.getRadius();
+          }
+          if (lineString) {
+            return lineString.getLength();
+          } else {
+            return 0;
+          }
+        };
+
+        this.getArea = function(geom, calculateLineStringArea) {
+          if (calculateLineStringArea && geom instanceof ol.geom.LineString) {
+            return Math.abs(new ol.geom.Polygon([geom.getCoordinates()]).
+                getArea());
+          } else if (geom instanceof ol.geom.LinearRing ||
+              geom instanceof ol.geom.Polygon) {
+            return Math.abs(geom.getArea());
+          } else if (geom instanceof ol.geom.Circle) {
+            return Math.PI * Math.pow(geom.getRadius(), 2);
+          }
+          return 0;
+        };
+
+        this.getAzimuth = function(geom) {
+          if (!(geom instanceof ol.geom.Polygon) &&
+              !(geom instanceof ol.geom.LineString) &&
+              !(geom instanceof ol.geom.LinearRing)) {
+            return 0;
+          }
+          var pt1, pt2, coords = geom.getCoordinates();
+          if (geom instanceof ol.geom.Polygon) {
+            coords = coords[0];
+          }
+          pt1 = coords[0];
+          pt2 = coords[1];
+
+          if (!pt1 || !pt2) {
+            return 0;
+          }
+          var x = pt2[0] - pt1[0];
+          var y = pt2[1] - pt1[1];
+          var rad = Math.acos(y / Math.sqrt(x * x + y * y));
+          var factor = x > 0 ? 1 : -1;
+          return (360 + (factor * rad * 180 / Math.PI)) % 360;
+        };
+
+        this.getLengthLabel = function(geom) {
+          return measureFilter(this.getLength(geom));
+        };
+
+        this.getAreaLabel = function(geom, calculateLineStringArea) {
+          return measureFilter(this.getArea(geom, calculateLineStringArea),
+              'area');
+        };
+
+        this.getAzimuthLabel = function(geom) {
+          return measureFilter(this.getAzimuth(geom), 'angle');
+        };
+
```

Comment:
The functions from here on out _could_ be put into separate service as they do different things the the ones above.